+Base/QRconv.m

```diff
@@ -0,0 +1,384 @@
+                        
+                        if dist_ < dist 
+                            CX(ii) = CX_;
+                            CY(ii) = CY_;
+                            isQR(ii) = true;
+                            dist = dist_;
+                        end
+                    end
+                end
+            end
+            
+            m_vectors(:,ii) = m(:);
+            
+            if CX(ii) == 0 && CY(ii) == 0   % Empty bits reads as [0,0] QR, so most [0,0] are false positives.
+                isQR(ii) = false;
+            end
+        end
+    end
+
+    CX(~isQR) = NaN;
+    CY(~isQR) = NaN;
+end
+function [CX, CY, version, checksum0] = interpretQR(m)
+    % From the code contained in m, attempt to read information.
+
+    length = 25;    % Total length of code
+    pad = [1 6];    % Pad locations of bits (indexed from 1)
+    vb = 4;         % Version bits
+    rb = 8;         % Number of bits to encode the row
+    cb = 8;         % Number of bits to encode the col
+    cs = 3;         % Checksum
+
+    assert(numel(m) == length, 'm is the wrong size')
+
+    if size(m, 1) > 1
+        m = m';
+    end
+
+    b = 2 .^ (0:7);
+    m(pad) = [];
+    p = 1;
+
+    version =   sum(m(p:p+vb-1) .* b(vb:-1:1)); p = p + vb;
+    CY =        sum(m(p:p+rb-1) .* b(rb:-1:1)); p = p + rb;
+    CX =        sum(m(p:p+cb-1) .* b(cb:-1:1)); p = p + cb;
+    checksum = 	sum(m(p:p+cs-1) .* b(cs:-1:1));
+
+    % Remove checksum, and test
+    m(end-cs+1:end) = [];
+
+    checksum0 = false;
+    if ~isempty(checksum)
+        checksum0 = mod(sum(m), 2^cs) == checksum;
+    end
+    
+    checksum0 = checksum0 && CX <= 120 && CY <= 120;
+end
+function cir = circleFunc(XX, YY, x0, y0, r)
+    cir = (XX - x0).^2 + (YY - y0).^2 < r^2;
+end
+function [M, b, M2, b2, outliers] = majorityVoteCoordinateFit(v, V, options_guess)
+    c = cos(options_guess.ang);
+    s = sin(options_guess.ang);
+    M_guess = [[s, c]; [-c, s]] / options_guess.calibration * options_guess.d;
+    
+    % The positions and labels of candidate QR codes define candidate coordinate systems.
+    % We want to find which candidate is correct.
+    b_guesses = v - M_guess * V;
+    
+    duplicates = false(1, size(b_guesses,2));
+    
+    for ii = 1:size(V,2)
+        dduplicates = V(1,:) == V(1, ii) & V(2,:) == V(2, ii);
+        
+        if sum(dduplicates) > 1
+            duplicates = duplicates | dduplicates;
+        end
+    end
+    
+    if isfield(options_guess, 'X_expected') && isfield(options_guess, 'Y_expected')
+        V_expected = [options_guess.X_expected; options_guess.Y_expected];
+    else
+        V_expected = [NaN; NaN];
+    end
+    b_expected = [options_guess.img_W; options_guess.img_H]/2 - M_guess * V_expected;
+    
+    % Setup variables that we will change as we loop.
+    mostvotes = 1;
+    b_guess = [NaN; NaN];
+    outliers = true(1, size(b_guesses,2));
+    dist = 3*options_guess.d;
```

Comment:
Why 3?

Reply:
Any jump larger than 3 QR regions should have to undergo further checks. The 3 itself was arbitrary.

Reply:
Feel free to add this enhancement.